File: SecureAndroidClient/app/src/main/java/com/security/crypto/KeyManager/KeyManagerImp.java
package com.security.crypto.KeyManager;

import java.io.IOException;
import java.security.PublicKey;
import java.security.cert.X509Certificate;

import javax.crypto.spec.SecretKeySpec;

public abstract class KeyManagerImp {


    //unfortunately this works only for Windows os
    public String Server_PUBLIC_KEY = "Public.key";
    public String Server_Certificate = "Certificate.pem";

    public abstract void SaveServerPublicKey();

    public abstract void SaveCertificate(String CertPemFormat);

    public abstract void ProduceCipherKey(String SessionResult);

    public abstract void ProduceIntegrityKey(String SessionResult);

    public abstract PublicKey loadRemoteServerPublicKey();

    public abstract X509Certificate loadCertificate() throws IOException;

    public abstract String loadRemoteCipherKey();

    public abstract SecretKeySpec loadRemoteIntegrityKey();


}

File: PythonClient/KeyManager/KeyManagerImp.py
import abc


class KeyManager(object):
    __metaclass__ = abc.ABCMeta

    @abc.abstractmethod
    def __SaveServerPublicKey(self):
        return

    @abc.abstractmethod
    def __SaveCertificate(self, CertPemFormat):
        return

    @abc.abstractmethod
    def __ProduceCipherKey(self, CertPemFormat):
        return

    @abc.abstractmethod
    def __ProduceIntegrityKey(self, SessionResult):
        return

    @abc.abstractmethod
    def __loadRemoteServerPublicKey(self):
        return

    @abc.abstractmethod
    def __loadCertificate(self):
        return

    @abc.abstractmethod
    def __loadRemoteCipherKey(self):
        return

    @abc.abstractmethod
    def __loadRemoteIntegrityKey(self):
        return
